shared/src/commonMain/kotlin/co/touchlab/kampstarter/models/BreedModel.kt

```diff
@@ -8,77 +8,61 @@ import co.touchlab.kampstarter.sqldelight.asFlow
 import co.touchlab.kermit.Kermit
 import co.touchlab.stately.ensureNeverFrozen
 import com.russhwolf.settings.Settings
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.launch
-import org.koin.core.KoinComponent
-import org.koin.core.get
 import org.koin.core.inject
 import org.koin.core.parameter.parametersOf
 
 class BreedModel(
-    private val viewUpdate: (ItemDataSummary) -> Unit,
     private val errorUpdate: (String) -> Unit
```

Comment:
I don't really like still having the error callback here. Kinda mixes strategies. For this use, the only error is the result of `getBreedsFromNetwork` So I think just having getBreedsFromNetwork return an optional error string would work

Reply:
makes sense, updated